dapp/src/components/buySell/BalanceHeader.js

```diff
@@ -233,26 +139,44 @@ const BalanceHeader = () => {
                   </span>
                 </>
               ) : (
-                '--.----'
+                '0'
```

Comment:
Sorry @ZaHuPro that we've been slow to merge this. 😓 I resolved the conflicts and made some small tweaks. I do think @sparrowDom should take a look at this change where the placeholder value is now `'0'`. Based on my experience running this branch, we don't seem to be distinguishing between "not yet loaded" and "not zero". Once the value is loaded, I think we would prefer the "0.00" or "0.000000" that would come from `displayedBalance`. Until then, I think we prefer to show the placeholder "--.----". It's not a big deal as-is for users to see "0" either way, but I think some @sparrowDom's other branches also have considerations for loading/disconnected states.

Reply:
Let me know if I have to revert this change

Reply:
I have reverted this change